src/features/todosState.ts

```diff
@@ -1,15 +1,84 @@
-import { createAsyncThunk, createSelector, createSlice } from '@reduxjs/toolkit'
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'
 
 import type { RootState } from '../app/store'
 import { todosController } from '../networking/controllers/todos'
-import { Todo } from '../types/globalTypes'
+import { NewItem, Todo } from '../types/globalTypes'
 
 export interface TodosState {
   todos: Todo[]
   loading: boolean
   error: string | undefined
 }
 
+export interface ItemCheck {
+  id: string
+  title: string
+  description: string
+  toCheck: boolean
+}
+
+export const clearTodoAPI = createAsyncThunk<
+  void,
+  string,
+  { rejectValue: string }
+>('todos/deleteTodosAPI', async (id: string, thunkAPI) => {
+  try {
+    return await todosController.deleteTodo(id)
+  } catch (error) {
+    if (
+      typeof error === 'object' &&
+      error &&
+      'message' in error &&
+      typeof error.message === 'string'
+    ) {
+      return thunkAPI.rejectWithValue(error.message)
+    } else {
+      return thunkAPI.rejectWithValue(JSON.stringify(error))
+    }
+  }
+})
+
+export const manageCheck = createAsyncThunk<
+  Todo,
+  ItemCheck,
+  { rejectValue: string }
+>('todos/manageCheckTodo', async (itemCheck: ItemCheck, thunkAPI) => {
+  try {
+    return await todosController.modifyTodo(itemCheck)
+  } catch (error) {
+    if (
+      typeof error === 'object' &&
+      error &&
+      'message' in error &&
+      typeof error.message === 'string'
+    ) {
+      return thunkAPI.rejectWithValue(error.message)
+    } else {
+      return thunkAPI.rejectWithValue(JSON.stringify(error))
+    }
```

Comment:
I see this is being repeated, maybe you can create a helper function for it, what do you think?